datafusion/core/tests/parquet_pruning.rs

```diff
@@ -449,6 +450,154 @@ async fn prune_int32_eq_in_list_negated() {
     assert_eq!(output.result_rows, 19, "{}", output.description());
 }
 
+async fn test_prune_decimal(
```

Comment:
I'm inspired by this, I will collation other test case and make the test cases cleanly

Reply:
Can you please clarify what you mean by this comment?

I didn't quite understand if you planned to do this combination as part of this PR or if you planned to do it as a follow on PR. 

In other words, is this PR ready for review, or do you plan more work?

Reply:
Sorry for the confused comments.
When I work on the decimal test, and find all the `pruning test` has the common logic.
I can collation them in the next pr.
```
async test_prune(
    case_type: Scenario,
    sql: &str,
    expected_errors: Option<usize>,
    expected_row_group_pruned: Option<usize>,
    expected_results: usize) -> {}
```
All the test cases such as `prune_f64_lt`,`prune_i64_lt` can use `test_prune` to replace the repeated logic.

Reply:
follow up pr: https://github.com/apache/arrow-datafusion/pull/2986